src/auth/src/credentials/idtoken.rs

```diff
@@ -12,10 +12,49 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+//! Obtain [OIDC ID Tokens].
+//!
+//! `IDTokenCredentials` provide a way to obtain OIDC ID tokens, which are
+//! commonly used for [service to service authentication], like when services are
+//! hosted in Cloud Run or mediated by Identity-Aware Proxy (IAP).
+//! Unlike access tokens, ID tokens are not used to authorize access to
```

Comment:
New idea or topic, new paragraph:

```suggestion
//! hosted in Cloud Run or mediated by Identity-Aware Proxy (IAP).
//!
//! Unlike access tokens, ID tokens are not used to authorize access to
```